src/background/wallet/unlock-wallet.ts

```diff
@@ -0,0 +1,32 @@
+import { decryptMnemonic } from '@background/crypto/mnemonic-encryption';
+import { getWallet } from './wallet-utils';
+
+export async function unlockWallet(
```

Comment:
Would consider renaming this method. "Unlocking the wallet" is what we tell the user, but as far as this method goes, I'd use something more technically descriptive. `getDecryptedWalletDetails`?